docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using OAuth2 RBAC
+
+include::./attributes.adoc[]
+:extension-name: Elytron Security OAuth2
+:oauth2-token: OAuth2 tokens
+
+This guide explains how your Quarkus application can utilize {oauth2-token} to provide secured access to the JAX-RS endpoints.
+
+If your OAuth2 Authentication server provides JWT tokens, you should use link:jwt-guide.html[MicroProfile JWT RBAC] instead, this extension aims to be used with obscure tokens and validate the token by calling an introspection endpoint.
+
+== Configuration
+
+[cols="<m,<m,<2",options="header"]
+|===
+|Property Name|Default|Description
+|quarkus.oauth2.enabled|true|Determine if the OAuth2 extension is enabled. Enabled by default if you include the elytron-security-oauth2 dependency, so this would be used to disable it.
```

Comment:
backticks around the dependency name would be nice.

Maybe also around the properties, depending on how it is done in the other guides.

Reply:
done.

backticks around properties when use a table is not the common usage on the other guides.